python/sbp/client/util/settingmonitor.py

```diff
@@ -36,5 +56,72 @@ def wait_for_setting_value(self, section, setting, value, wait_time=5.0):
         return ok
 
     def clear(self):
+        """Clear all settings"""
         self.settings[:] = []
 
+
+
+if __name__ == "__main__":
+    from sbp.client import Handler, Framer
+    from sbp.client.drivers.network_drivers import TCPDriver
+    from sbp.settings import MsgSettingsWrite, MsgSettingsReadReq
+    import argparse
+
+    def print_setting(sbp_msg, **metadata):
+        print sbp_msg
+
+    def main():
+        parser = argparse.ArgumentParser(
+            description="Swift Navigation SBP Setting Monitor example.")
+        parser.add_argument(
+            "-H",
+            "--host",
+            required=True,
+            help="specify the host address.")
+        parser.add_argument(
+            "-p",
+            "--port",
+            default=55556,
+            help="specify the port to use.")
+        args = parser.parse_args()
+
+        monitor = SettingMonitor()
+
+        with TCPDriver(args.host, args.port) as driver:
+            with Handler(Framer(driver.read, driver.write, verbose=True)) as link:
+                driver.flush()
+                time.sleep(2)
+
+                # Capture setting messages
+                link.add_callback(monitor.capture_setting,
+                                  SBP_MSG_SETTINGS_READ_RESP)
+
+                link.add_callback(print_setting, SBP_MSG_SETTINGS_READ_RESP)
+
+                # Disable spectrum analyzer
+                link(MsgSettingsWrite(setting='%s\0%s\0%s\0' % (
```

Comment:
It might be worth providing utility functions for handling the NULL deliminated settings.

Reply:
I agree. The same pattern shows up in the Console's settings view and I'm guessing other places, too. I think it might be better to do this as a separate task once we identify other places where this pattern is used.

Reply:
If/when we redesign settings I'd like to use a different setting deliminator.

Reply:
110% support for different deliminator. C string functions don't work well with the current one.